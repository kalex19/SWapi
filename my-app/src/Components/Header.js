import React, {Component} from 'react';
import Styles from '.Header.scss';

export default Header from Component{

const ...

//favorite functionality
//const Hello = ({name}) => <div>Hello {name}</div>
render () {
    return(
      <div>
        <header>
          <h1>Star Wars</h1>
          <input>Favorites</input
        </header>
      </div>
    ) 
  }
}


proptypes: