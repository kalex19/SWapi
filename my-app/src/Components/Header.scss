@import '../variables.scss';

.header {
  display: flex;
  justify-content: space-between;
}

.title {
  color: $main-color;
  font-size: 5em;
  margin: 16px;
  text-align: center;
  text-shadow: $main-shadow 2px 1px;
}

.favBtn {
  background-color: $main-shadow;
  border: 2px $main-color solid;
  color: #000;
  font-size: 1em;
  height: 30px;
  margin-top: 6%;
  opacity: 0.8;
  text-align: center;
  width: 19%;
}