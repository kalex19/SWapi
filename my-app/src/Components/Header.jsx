import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Header.scss';

const Header = props => {
	return (
		<div className="header">
			<h1 className="title"> SWapi-Box </h1>
			<p className="favBtn" role="button">
				View Favorites ({props.favoriteCount})
			</p>
		</div>
	);
};

Header.propTypes = {
	favoriteCount: PropTypes.number
};

export default Header;
