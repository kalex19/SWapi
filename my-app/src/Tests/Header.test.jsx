import React from 'react';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import Header from '../Components/Header';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Header', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<Header />);
	});

	it('should mock snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('', () => {});

	it('', () => {});
});
