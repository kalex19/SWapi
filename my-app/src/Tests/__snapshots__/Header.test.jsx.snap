// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should mock snapshot 1`] = `undefined`;
